File: Simple-E-Commerce/Cheese.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Simple_E_Commerce
{
    class Cheese : Product , IShippable , IExpirable
    {
        public DateTime ExpireTime { get;  }
        public double Weight { get; }

        public Cheese(string name , double price , int quantity , DateTime expireTime , double weight)
            :base(name , price , quantity)
        {
            ExpireTime = expireTime;
            Weight = weight; 
        }

        public string GetName()
        {
            return Name ; 
        }

        public double GetWeight()
        {
            return Weight;
        }

        public bool IsExpired()
        {
            return DateTime.Now > ExpireTime;
        }
    }
}

File: Using_Java/Simple-E-Commerce/src/Cheese.java
import java.time.LocalDateTime;

public class Cheese extends Product implements IShippable, IExpirable {
    private LocalDateTime expireTime;
    private double weight;

    public Cheese(String name, double price, int quantity, LocalDateTime expireTime, double weight) {
        super(name, price, quantity);
        this.expireTime = expireTime;
        this.weight = weight;
    }

    @Override
    public String getName() {
        return super.getName();
    }

    @Override
    public double getWeight() {
        return weight;
    }

    @Override
    public boolean isExpired() {
        return LocalDateTime.now().isAfter(expireTime);
    }
} 